File: core_algorithms/knapsackLight/knapsackLight.py
def knapsackLight(value1, weight1, value2, weight2, maxW):
    if maxW >= weight1 + weight2:
        return value1 + value2
    if maxW < weight1 and maxW < weight2:
        return 0
    if maxW < weight1 and maxW >= weight2:
        return value2
    elif maxW < weight2 and maxW >= weight1:
        return value1
    else:
        return max(value1, value2)

File: intro_algorithms/knapsackLight/knapsackLight.ts
function knapsackLight(value1: number, weight1: number, value2: number, weight2: number, maxW: number): number {
    if(maxW >= weight1 + weight2) {
        return value1 + value2;
    }
    if(canCarry(weight1, maxW) && canCarry(weight2, maxW)) {
        return Math.max(value1, value2);
    }
    if(canCarry(weight1, maxW) && !canCarry(weight2, maxW)) {
        return value1;
    }
    if(canCarry(weight2, maxW) && !canCarry(weight1, maxW)) {
        return value2;
    }
    else {
        return 0;
    }

}

function canCarry(weight: number, max: number) {
    return max >= weight;
}